src/main/java/command/AddCommand.java

```diff
@@ -0,0 +1,66 @@
+package command;
+
+import java.time.LocalDate;
+
+import exception.LunaException;
+import storage.Storage;
+// Import Tasks
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+import ui.Ui;
+
+/**
+ * Encapsulates a user instruction to create a new task.
+ *
+ * @author fannyjian
+ */
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Initialises a command to create a new Todo.
+     *
+     * @param des Description of task.
+     */
+    public AddCommand(String des) {
+        this.task = new Todo(des);
+    }
+
+    /**
+     * Initialises a command to create a new Deadline or Event.
+     *
+     * @param type Type of Task.
+     * @param des Description of Task.
+     * @param date Date to be completed by.
+     */
+    public AddCommand(String type, String des, LocalDate date) {
```

Comment:
I like how you have used OOPs principles to distinguish between todo and deadline and event tasks.